src/main/java/com/google/sps/servlets/queue/RemoveFromQueue.java

```diff
@@ -73,16 +80,34 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
                     .findFirst()
                     .orElse(null);
 
-            // Delete workspace
-            String workspaceID = (String) delEntity.getProperty("workspaceID");
-            factory.fromWorkspaceID(workspaceID).delete();
+            if (delEntity != null) {
```

Comment:
A clearer way to do this would be to remove `.orElse(null)` on line `81` and then check `delEntity.isPresent()` on the `Optional`. `delEntity` would then become an `Optional<EmbeddedEntity>` instead of an `EmbeddedEntity`.

Reply:
Another option would be to do 
`delEntity.ifPresent((entity) -> {code to run if entity is present});`